versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/VersionStoreImpl.java

```diff
@@ -962,28 +973,41 @@ static IdentifiedContentKey buildIdentifiedKey(
   }
 
   @Override
-  public Map<ContentKey, ContentResult> getValues(Ref ref, Collection<ContentKey> keys)
+  public Map<ContentKey, ContentResult> getValues(
+      Ref ref, Collection<ContentKey> keys, boolean returnNotFound)
       throws ReferenceNotFoundException {
     RefMapping refMapping = new RefMapping(persist);
     CommitObj head = refMapping.resolveRefHead(ref);
-    if (head == null) {
-      return emptyOrNotFound(ref, emptyMap());
-    }
 
     try {
       IndexesLogic indexesLogic = indexesLogic(persist);
-      StoreIndex<CommitOp> index = indexesLogic.buildCompleteIndex(head, Optional.empty());
+      StoreIndex<CommitOp> index =
+          head != null
+              ? indexesLogic.buildCompleteIndex(head, Optional.empty())
+              : emptyImmutableIndex(COMMIT_OP_SERIALIZER);
 
       ContentMapping contentMapping = new ContentMapping(persist);
-      return contentMapping.fetchContents(index, keys).entrySet().stream()
-          .collect(
-              Collectors.toMap(
-                  Map.Entry::getKey,
-                  e ->
-                      contentResult(
-                          buildIdentifiedKey(e.getKey(), index, e.getValue(), x -> null),
-                          e.getValue(),
-                          null)));
+      Map<ContentKey, ContentResult> result =
+          contentMapping.fetchContents(index, keys).entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      Map.Entry::getKey,
+                      e ->
+                          contentResult(
+                              buildIdentifiedKey(e.getKey(), index, e.getValue(), x -> null),
+                              e.getValue(),
+                              null)));
+      if (returnNotFound) {
+        keys.stream()
+            .filter(k -> !result.containsKey(k))
+            .forEach(
+                key -> {
+                  IdentifiedContentKey identifiedKey =
+                      buildIdentifiedKey(key, index, null, null, x -> null);
+                  result.put(key, contentResult(identifiedKey, null, null));
```

Comment:
1. This is relying on `ContentMapping.fetchContents` returning a mutable map.
2. Wouldn't it be more efficient to pass `returnNotFound` down to `fetchContents`?

Reply:
Not sure, but the code here can be refactored.

Reply:
I think, it looks easier w/o the Java stream dance